Sources/MapboxNavigation/NavigationMapView+Annotations.swift

```diff
@@ -0,0 +1,488 @@
+        }
+        
+        if style.image(withId: ImageIdentifier.trafficSignalNight) == nil,
+           let trafficSignalNight =  Bundle.mapboxNavigation.image(named: "TrafficSignalNight") {
+            try style.addImage(trafficSignalNight, id: ImageIdentifier.trafficSignalNight)
+        }
+        
+        if style.image(withId: ImageIdentifier.railroadCrossingDay) == nil,
+           let railroadCrossingDay =  Bundle.mapboxNavigation.image(named: "RailroadCrossingDay") {
+            try style.addImage(railroadCrossingDay, id: ImageIdentifier.railroadCrossingDay)
+        }
+        
+        if style.image(withId: ImageIdentifier.railroadCrossingNight) == nil,
+           let railroadCrossingNight =  Bundle.mapboxNavigation.image(named: "RailroadCrossingNight") {
+            try style.addImage(railroadCrossingNight, id: ImageIdentifier.railroadCrossingNight)
+        }
+    }
+    
+    func updateIntersectionSignals(with routeProgress: RouteProgress) {
+        guard !routeProgress.routeIsComplete else {
+            removeIntersectionSignals()
+            return
+        }
+        var featureCollection = FeatureCollection(features: [])
+        
+        let stepProgress = routeProgress.currentLegProgress.currentStepProgress
+        let intersectionIndex = stepProgress.intersectionIndex
+        let stepIntersections = stepProgress.intersectionsIncludingUpcomingManeuverIntersection
+        
+        for intersection in stepIntersections?.suffix(from: intersectionIndex) ?? [] {
+            if let feature = signalFeature(from: intersection, styleType: styleType) {
+                featureCollection.features.append(feature)
+            }
+        }
+        
+        let style = mapView.mapboxMap.style
+        
+        do {
+            let sourceIdentifier = NavigationMapView.SourceIdentifier.intersectionSignalSource
+            if style.sourceExists(withId: sourceIdentifier) {
+                try style.updateGeoJSONSource(withId: sourceIdentifier, geoJSON: .featureCollection(featureCollection))
+            } else {
+                var source = GeoJSONSource()
+                source.data = .featureCollection(featureCollection)
+                try style.addSource(source, id: sourceIdentifier)
+            }
+            
+            let layerIdentifier = NavigationMapView.LayerIdentifier.intersectionSignalLayer
+            guard !style.layerExists(withId: layerIdentifier) else { return }
+            
+            var shapeLayer = SymbolLayer(id: layerIdentifier)
+            shapeLayer.source = sourceIdentifier
+            shapeLayer.iconAllowOverlap = .constant(false)
+            shapeLayer.iconImage = .expression(Exp(.get) {
+                "imageName"
+            })
+            
+            let layerPosition = layerPosition(for: layerIdentifier)
+            try style.addPersistentLayer(shapeLayer, layerPosition: layerPosition)
```

Comment:
Looks like the intersection annotation layer is on top of the maneuver arrow layer. That makes logical sense and maintains the illusion of these traffic lights and signs standing upright. However, the maneuver arrow is an important element to be obscuring with an icon.

Will the maneuver arrow be prominent enough to overcome this overlap? Or should be consider a visual mitigation, such as offsetting the traffic light at a maneuver point?

Reply:
I think we could solve this issue by having a smaller icons or a longer maneuver arrow, from a UX design perspective. Because it's easy to understand by having the traffic signal icons place above the route line, which means that the user will drive through them.